students/grzegorz_kowalik/lesson_03_functions/collatz.py

```diff
@@ -0,0 +1,15 @@
+def collatz(number):
+    print(number)
+
+    if number == 1:
+        print("Koniec")
+    elif number % 2 == 0:
+        collatz(number // 2)
+    else:
+        collatz(3 * number + 1)
+
+
```

Comment:
całą tą część kodu, która już wywołuje kalkulator można by zamknąć w warunku:
`if name == 'main':`
Dzięki czemu podczas importu funkcji z tego modułu do innego modułu/skryptu ta część modułu nie zostanie wykonana.
On się wykona tylko wtedy, gdy ten moduł/skrypt zostanie uruchomiony jak skrypt (na najwyższym poziomie).
Jeżeli uruchomisz ten skrypt zmienna name przyjmie wartość 'main', jeżeli zaimportujesz jako moduł to name ma wartość: 'nazwa_pliku'.
Pamiętaj o tym tworząc nowy moduł i możesz to tez dodać do pozostałych zadań.
W skryptach z lekcji 0 uzyty był ten mechanizm, możesz zobaczyć jak go użyto.